src/services/schemas/utilities.ts

```diff
@@ -231,15 +231,23 @@ export function getSchemaValueDefinitionIndex(definitions: Schema[], value: Sche
     return definitions.findIndex(definition => definition.type === 'block')
   }
 
-  switch (typeof value) {
+  let parsedValue = value
+
+  try {
+    parsedValue = parseUnknownJson(value)
+  } catch {
+    // do nothing, use the raw value
+  }
```

Comment:
Isn't this going to mess up some types that could be parsed as json but are actually just supposed to be strings?

Reply:
For posterity: after chatting we've decided to add a check on the `parsedValue`'s type as well and only continue with objects so as not to unintentionally parse strings that contain numbers, booleans, etc